Proteomics/Modifications/ModificationLocalization.cs

```diff
@@ -16,29 +16,32 @@ public static bool ModFits(ModificationWithMass attemptToLocalize, Protein prote
             while (indexUp < motif.ToString().Length)
             {
                 if (indexUp + proteinToMotifOffset < 0 || indexUp + proteinToMotifOffset >= protein.Length
-                    || (!char.ToUpper(motif.ToString()[indexUp]).Equals('X') 
+                    || (!char.ToUpper(motif.ToString()[indexUp]).Equals('X')
                         && !char.ToUpper(motif.ToString()[indexUp]).Equals(protein.BaseSequence[indexUp + proteinToMotifOffset])))
                 {
                     return false;
                 }
                 indexUp++;
             }
-            if (attemptToLocalize.terminusLocalization == TerminusLocalization.NProt && proteinOneBasedIndex > 2)
+            if (attemptToLocalize.Position == "N-terminal." && proteinOneBasedIndex > 2)
             {
                 return false;
             }
-            if (attemptToLocalize.terminusLocalization == TerminusLocalization.NPep && peptideOneBasedIndex > 1)
+            if (attemptToLocalize.Position == "Peptide N-terminal." && peptideOneBasedIndex > 1)
             {
                 return false;
             }
-            if (attemptToLocalize.terminusLocalization == TerminusLocalization.PepC && peptideOneBasedIndex < peptideLength)
+            if (attemptToLocalize.Position == "C-terminal." && peptideOneBasedIndex < peptideLength)
             {
                 return false;
             }
-            if (attemptToLocalize.terminusLocalization == TerminusLocalization.ProtC && proteinOneBasedIndex < protein.Length)
+            if (attemptToLocalize.Position == "Peptide C-terminal." && proteinOneBasedIndex < protein.Length)
             {
                 return false;
             }
+
+            // I guess Anywhere. and Unassigned. are true since how do you localize anywhere or unassigned.
```

Comment:
I realize this is nitpicky but this comment is confusing because we're using "Anywhere." as "not specific to a terminus". It is completely reasonable to localize a modification to a phosphorylation on a serine in the middle of a peptide.

Reply:
that comment is because I don't understnad the whole code block and what it's trying to do. would you help me? I would like to add the description to the top of the code block then handle this correctly.